public/app/features/dashboard/components/DashExportModal/DashboardExporter.test.ts

```diff
@@ -94,6 +94,41 @@ it('handles a default datasource in a template variable', async () => {
   expect(exported.templating.list[0].datasource.uid).toBe('${DS_GFDB}');
 });
 
+it('do not expose datasource name and id in a in a template variable of type datasource', async () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I would not add this line. Instead, put the type in the dashboard
`const dashboard: Dashboard`. You just need to add some extra properties such as 
```
    title: 'My dashboard',
    revision: 1,
    editable: false,
    graphTooltip: DashboardCursorSync.Off,
    schemaVersion: 1,
    timepicker: { hidden: true },
    timezone: '',
    panels: [
      {
        id: 1,
        type: 'timeseries',
        title: 'My panel title',
        gridPos: { x: 0, y: 0, w: 1, h: 1 },
      },
    ],
```